src/rabbit_registry.erl

```diff
@@ -114,6 +116,29 @@ conditional_unregister({exchange_decorator, Type}) ->
 conditional_unregister(_) ->
     ok.
 
+% update channels to enable/disable interceptors
+registered({{Class, Type}, ModuleName}) ->
+    ClassModule = class_module(Class),
+    case proplists:get_value(registered, ClassModule:module_info(functions)) of
```

Comment:
I think this problem deserves a deeper solution. For example by creating a behaviour that's related to the `rabbit_registry`. Also not sure taking the common `registered` and `unregistered` names is a good idea, since many modules might have those functions and not be related to the rabbit registry.

@michaelklishin @dumbbell you should provide input here